src/relations/postgresql_provider.py

```diff
@@ -190,6 +199,13 @@ def update_endpoints(self, event: DatabaseRequestedEvent = None) -> None:
                 read_only_endpoints,
             )
 
+    def _check_multiple_endpoints(self) -> bool:
+        """Checks if there are relations with other endpoints."""
+        relation_names = [relation.name for relation in self.charm.client_relations]
+        if "database" in relation_names and len(relation_names) > 1:
+            return True
```

Comment:
Can this return `True` if there are multiple applications related to the `database` endpoint?

Reply:
I think the logic from `src/relations/db.py`'s `_check_multiple_endpoints` will help you here.